src/ralph/cli.py

```diff
@@ -91,41 +103,76 @@ def extract(parser, chunksize):
         "Extracting events using the %s parser (chunk size: %d)", parser, chunksize
     )
 
-    parser = get_class_from_name(parser, Parsers)()
+    parser = get_class_from_name(parser, PARSERS)()
 
     for event in parser.parse(sys.stdin, chunksize=chunksize):
         click.echo(event)
 
 
-@click.argument("archive")
-@backends_options(backends=STORAGE_BACKENDS)
-def fetch(backend, archive, **options):
-    """Fetch an archive from a configured storage backend"""
+@click.argument("archive", required=False)
+@backends_options(backends=BACKENDS)
+@click.option(
+    "-c",
+    "--chunk-size",
+    type=int,
+    default=DEFAULT_BACKEND_CHUNCK_SIZE,
+    help="Get events by chunks of size #",
+)
+def fetch(backend, archive, chunk_size, **options):
+    """Fetch an archive or records from a configured backend"""
 
-    logger.info("Fetching archive %s from the configured %s backend", archive, backend)
+    logger.info(
+        "Fetching data from the configured %s backend (archive: %s | chunk size: %s)",
+        backend,
+        archive,
+        chunk_size,
+    )
     logger.debug("Backend parameters: %s", options)
 
-    storage = get_instance_from_class(
-        get_class_from_name(backend, StorageBackends), **options
-    )
-    storage.read(archive)
+    klass = get_class_from_name(backend, BACKENDS)
```

Comment:
the typo is intended ? `class` is a reserved name but you can maybe found something more relevant ?

Reply:
This is not a typo :sweat_smile: I'll make it more explicit.